rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -562,15 +556,45 @@ impl PendingMessage {
                 0
             }
         };
+        // Skip this message if it has been retried too many times
+        if let Some(max_retries) = retries_before_skipping {
+            if num_retries >= max_retries {
+                return None;
+            }
+        }
+        Some(num_retries)
+    }
 
-        let mut pm = Self::new(message, ctx, message_status, app_context);
-        if num_retries > 0 {
-            let next_attempt_after =
-                PendingMessage::calculate_msg_backoff(num_retries).map(|dur| Instant::now() + dur);
-            pm.num_retries = num_retries;
-            pm.next_attempt_after = next_attempt_after;
+    fn get_message_status(
+        origin_db: Arc<dyn HyperlaneDb>,
+        message: HyperlaneMessage,
```

Comment:
This method does not need to take ownership of message. I wonder if passing as a reference will be sufficient.